rs/tests/boundary_nodes/rate_limit_canister_test.rs

```diff
@@ -1,403 +1,303 @@
-    let agent = Agent::builder()
-        .with_url(domain)
-        .with_identity(identity)
-        .build()
-        .expect("failed to build the agent");
-    agent.fetch_root_key().await.unwrap();
-    agent
-}
-
-async fn add_config_1(logger: Logger, agent: &Agent, canister_id: Principal) {
-    // Note two rules (indices = [0, 2]) are linked to the same incident_id_1
-    // RuleIds are generated randomly on the canister side
-    let rule_1 = RateLimitRule {
-        canister_id: Some(canister_id),
-        subnet_id: None,
-        methods_regex: Some(Regex::new(r"^(method_1)$").unwrap()),
-        request_types: Some(vec![RequestType::Call]),
-        limit: Action::Limit(1, Duration::from_secs(60)),
-    };
-
-    let rule_2 = RateLimitRule {
-        canister_id: Some(canister_id),
-        subnet_id: None,
-        methods_regex: Some(Regex::new(r"^(method_2)$").unwrap()),
-        request_types: Some(vec![RequestType::Query]),
-        limit: Action::Limit(2, Duration::from_secs(60)),
-    };
-
-    let rule_3 = RateLimitRule {
-        canister_id: Some(canister_id),
-        subnet_id: None,
-        methods_regex: Some(Regex::new(r"^(method_3)$").unwrap()),
-        request_types: None,
-        limit: Action::Limit(3, Duration::from_secs(60)),
-    };
-
-    let rule_4 = RateLimitRule {
-        canister_id: Some(canister_id),
-        subnet_id: None,
-        methods_regex: Some(Regex::new(r"^(method_4)$").unwrap()),
-        request_types: Some(vec![RequestType::ReadState]),
-        limit: Action::Limit(4, Duration::from_secs(60)),
-    };
-
-    let args = Encode!(&InputConfig {
-        schema_version: 1,
-        rules: vec![
-            InputRule {
-                incident_id: "b97730ac-4879-47f2-9fea-daf20b8d4b64".to_string(),
-                rule_raw: rule_1.to_bytes_json().unwrap(),
-                description:
-                    "Some vulnerability #1 discovered, temporarily rate-limiting the canister calls"
-                        .to_string(),
-            },
-            InputRule {
-                incident_id: "f63c821c-9320-476a-bc89-94cb99d04639".to_string(),
-                rule_raw: rule_2.to_bytes_json().unwrap(),
-                description: "Some vulnerability #2 discovered".to_string(),
-            },
-            InputRule {
-                incident_id: "b97730ac-4879-47f2-9fea-daf20b8d4b64".to_string(),
-                rule_raw: rule_3.to_bytes_json().unwrap(),
-                description:
-                    "Some vulnerability #1 discovered, temporarily rate-limiting the canister calls"
-                        .to_string(),
-            },
-            InputRule {
-                incident_id: "389bbff8-bffa-4430-bb70-8ce1ea399c07".to_string(),
-                rule_raw: rule_4.to_bytes_json().unwrap(),
-                description: "Some vulnerability #3 discovered".to_string(),
-            },
-        ],
+    let args = Encode!(&InitArg {
+        registry_polling_period_secs: 60,
```

Comment:
How about we set this a bit lower to like 5s just that the canister quickly sees all the BNs?

Reply:
We should be below 5mins for p90. ic-boundary alraedy "only" pulls every 30s. So, it adds up.

Reply:
sure, sounds good!

Reply:
we could also make `ic-boundary` polling more frequent than default `30s`

Reply:
Making it more frequent is harder, because it is part of the env file in the guest OS and cannot just be passed in like that. Let's try the 5s and then, we see.